android/app/src/main/java/com/zulipmobile/notifications/NotificationHelper.kt

```diff
@@ -97,21 +109,20 @@ fun extractTotalMessagesCount(conversations: ByConversationMap): Int {
 private fun buildKeyString(fcmMessage: MessageFcmMessage): String {
     val recipient = fcmMessage.recipient
     return when (recipient) {
-        is Recipient.Stream -> String.format("%s:%s:stream", fcmMessage.sender.fullName,
-            recipient.stream)
-        is Recipient.GroupPm -> String.format("%s:%s:group", fcmMessage.sender.fullName,
-            recipient.getPmUsersString())
-        else -> String.format("%s:%s:private", fcmMessage.sender.fullName,
-            fcmMessage.sender.email)
+        is Recipient.Stream -> String.format("%s:%s:stream", recipient.stream, recipient.topic)
```

Comment:
I can see that this should be a separate commit, and probably not part of this issue.  I was thinking that different topics in a stream should be different conversations, but this may be too fine of definition for a notification.  It would probably lead to numerous notifications for just one stream, an prevent others notifications from being shown.